tests/e2e/file_volume_statefulsets.go

```diff
@@ -0,0 +1,687 @@
+			"Number of Pods in the statefulset should match with number of replicas")
+	})
+
+	/*
+		Test performs following operations
+
+		Steps
+		1. Create a storage class.
+		2. Create nginx service.
+		3. Create nginx statefulsets with 3 replicas.
+		4. Wait until all Pods are ready and PVCs are bounded with PV.
+		5. Uninstall CSI & wait for all the pods to delete
+		6. Install CSI again & wait for all the pods to be up and running again
+		7. Scale down statefulsets to 2 replicas.
+		8. Scale up statefulsets to 3 replicas.
+		9. Scale down statefulsets to 0 replicas and delete all pods.
+		10. Delete all PVCs from the tests namespace.
+		11. Delete the storage class.
+	*/
+	ginkgo.It("Statefulset with file volume testing with CSI uninstall and install again", func() {
+		curtime := time.Now().Unix()
+		randomValue := rand.Int()
+		val := strconv.FormatInt(int64(randomValue), 10)
+		val = string(val[1:3])
+		curtimestring := strconv.FormatInt(curtime, 10)
+		scName := "nginx-sc-default-" + curtimestring + val
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+		scParameters[scParamFsType] = nfs4FSType
+		scSpec := getVSphereStorageClassSpec(scName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+		statefulset := GetStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		statefulset.Spec.VolumeClaimTemplates[len(statefulset.Spec.VolumeClaimTemplates)-1].Spec.AccessModes[0] =
+			v1.ReadWriteMany
+		statefulset.Spec.VolumeClaimTemplates[len(statefulset.Spec.VolumeClaimTemplates)-1].
+			Annotations["volume.beta.kubernetes.io/storage-class"] = scName
+		CreateStatefulSet(namespace, statefulset, client)
+		replicas := *(statefulset.Spec.Replicas)
+		// Waiting for pods status to be Ready
+		fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+		gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+		ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
+		gomega.Expect(ssPodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(),
+			fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+		gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset should match with number of replicas")
+
+		// Delete CSI Pods so that they auto restart
+		cmd = []string{"delete", "pods", "--all"}
```

Comment:
we should also delete node daemon-set pods and create them again. node daemon set restart is more important than CSI pod restart

Reply:
As discussed, this is an upgrade usecase and to be executed manually - removing the whole test